### PR TITLE
[WIP] IPv6

### PR DESCRIPTION
**Pull request by 1div0**
I'd like to ask you for a review of proposed changes for simultaneous support of IPv4 and IPv6.

Thank you.

### CODE REVIEW

**Comment by 1div0** [comment]
So far tried this:
€ srt-file-transmit /1TB/Video/HEVC/Main/LG\ Sydney\ 4K\ Demo.ts srt://ipv6.reflexion.tv:10000
IPv6 Unicast 2a02:c207:2019:8127::1 
Target disconnected


On the server side it looks like:
€ tshark -i Contabo udp
Capturing on 'Contabo'
    1 0.000000000 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    2 0.254012155 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    3 0.503958829 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    4 0.754957926 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    5 1.005103641 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    6 1.259036663 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    7 1.508955569 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    8 1.757038019 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
    9 2.009128179 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   10 2.258064017 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   11 2.513947139 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   12 2.764026134 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   13 890.784008053 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   14 891.034341547 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   15 891.289025295 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   16 891.539091055 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   17 891.789255451 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   18 892.039998209 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   19 892.284102403 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   20 892.534436626 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   21 892.797006106 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   22 893.044440487 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   23 893.299088950 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake
   24 893.549729387 2a02:c207:2019:8127::1080 → 2a02:c207:2019:8127::1 UDT 112 UDT type: handshake

File: testing/testmedia.cpp [hunk 1]
```diff
@@ -358,7 +358,7 @@ void SrtCommon::PrepareListener(string host, int port, int backlog)
         srt_conn_epoll = AddPoller(m_bindsock, SRT_EPOLL_OUT);
     }
 
-    sockaddr_in sa = CreateAddrInet(host, port);
+    sockaddr_in6 sa = CreateAddrInet(host, port);
```
Comment (maxsharabayko) [inline testing/testmedia.cpp]: Shouldn't it be `sockaddr_any` here instead of `sockaddr_in6`?

File: apps/apputil.cpp [hunk 2]
```diff
@@ -77,29 +77,85 @@ int inet_pton(int af, const char * src, void * dst)
...
-
-        sa.sin_addr = *(in_addr*)he->h_addr_list[0];
-    }
+        memset (&hints, 0, sizeof hints);
+        hints.ai_family = AF_UNSPEC;    // Allow IPv4 or IPv6
+        hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+        hints.ai_flags = 0;
+        hints.ai_protocol = 0;          // Any protocol
+
+        error = getaddrinfo(name.c_str(), port.c_str(), &hints, &destination);
```
Comment (ethouris) [inline apps/apputil.cpp]: A question: did you intend to support specifications like `myhost.com:complex-main` (as `myhost.com:5000`) ? Then this syntax is likely not supported by `UriParser`.

**Comment by maxsharabayko** [comment]
Hi @1div0.
Please consider the latest SRT master and srt-live-transmit. It now has support for IPv6 introduced by @ethouris.

**Comment by ethouris** [comment]
@1div0 : The changes you submitted here are probably provided from elsewhere and are already in the master. Please check for yourself if this is true, and if not, make changes here.

As we consider the problem fixed, this PR will be closed after 1 month of inactivity.

**Comment by 1div0** [comment]
OK, glad to see the progress on this topic.

Thanks. Cheers!

**Comment by ethouris** [comment]
#1381 addresses also some additional issues

**Comment by maxsharabayko** [comment]
This PR is likely outdated and no longer relevant as `sockaddr_any` is used in apps now, allowing both IPv4 and IPv6.

**Comment by 1div0** [comment]
Привет @maxsharabayko 

Feel free to close this PR.

Cheers,
Peter

**Pull request closed without merging**